tests/optional_dependency_test.py

```diff
@@ -0,0 +1,22 @@
+from unittest import mock
+import pytest
+import warnings
+
+
+def test_warning_raised_if_rdkit_not_installed():
+    #https://stackoverflow.com/questions/60746184/test-behavior-of-code-if-optional-module-is-not-installed
+    import sys
+
+    # clear cached modules
+    mods = list(sys.modules.keys())
+    for mod in mods:
+        if 'chemreps' in mod or 'rdkit' in mod:
```

Comment:
mods is the list, so i think we can remove the list over mods, and instead check :
```suggestion
    if chemreps' in mod or 'rdkit' in mods:
```